test/manipulation.test.js

```diff
@@ -357,6 +609,18 @@ describe('manipulation', function () {
       });
     });
 
+    it('should preserve properties with dynamic setters on create (promise variant)', function(done) {
+      StubUser.updateOrCreate({ id: 'newid', password: 'foo' })
+      .then(function(created) {
+        created.password.should.equal('foo-1');
+        return StubUser.findById(created.id)
+        .then(function(found) {
+          found.password.should.equal('foo-1');
+          done();
+        });
+      }).catch(done);
+    });
```

Comment:
"preserve properties with dynamic setter" is not related to callback vs. promise, can we omit this test?